src/megaapi_impl.cpp

```diff
@@ -5176,7 +5185,7 @@ MegaApiImpl::~MegaApiImpl()
     delete httpio;
 #endif
 
-    fireOnRequestFinish(request, MegaError(API_OK));
+    fireOnRequestFinish(request, new MegaErrorPrivate(API_OK));
```

Comment:
as noted above, the parameter here could be mega::make_unique<MegaErrorPrivate>(API_OK) to indicate ownership and have the compiler enforce it.

Reply:
Agree

Reply:
It's solved at Matt PR #2045